jax_rocm_plugin/pjrt/python/__init__.py

```diff
@@ -128,6 +130,54 @@ def set_rocm_paths(path):  # pylint: disable=too-many-branches
     os.environ["JAX_ROCM_PLUGIN_INTERNAL_LLD_PATH"] = lld_path
 
 
+def is_amd_gpu_available() -> bool:
+    """Checks if any AMD GPU is available to safeguard loading of the ROCm pjrt.
+
+    Different approaches to this are possible. We chose testing for existence
+    of KFD kernel driver entities as a proxy for the presence of AMD GPUs as
+    a good compromise between performance, reliability and simplicity.
+    Presence of such entities doesn't guarantee that the GPUs are usable through
+    HIP and PJRT, however, we can't do much much better without spawning an
+    additional process with a potentially complicated setup to run actual HIP
+    code. And we don't want to initialize HIP right now inside the current
+    process, because doing so might spoil a proper initialization of the
+    rocprofiler-sdk later during PJRT startup."""
+
+    try:
+        kfd_nodes_path = "/sys/class/kfd/kfd/topology/nodes/"
+        if not os.path.exists(kfd_nodes_path):
+            return False
+
+        r_wave_front_size = re.compile(r"\bwave_front_size\s+(\d+)\b", re.MULTILINE)
```

Comment:
nit: It'd be nice to have a comment explaining what we expect this regex is for and what it'll match.